src/usr/bin/diffuse.py.in

```diff
@@ -7181,30 +7182,6 @@ class Diffuse(Gtk.Window):
         def reload_file_cb(self, widget, data):
             self.open_file(self.current_pane, True)
 
-        # check changes to files on disk when receiving keyboard focus
-        def focus_in(self, widget, event):
-            for f, h in enumerate(self.headers):
-                info = h.info
-                try:
-                    if info.last_stat is not None:
-                        info = h.info
-                        new_stat = os.stat(info.name)
-                        if info.last_stat[stat.ST_MTIME] < new_stat[stat.ST_MTIME]:
-                            # update our notion of the most recent modification
-                            info.last_stat = new_stat
-                            if info.label is not None:
-                                s = info.label
-                            else:
-                                s = info.name
-                            msg = _('The file %s changed on disk.  Do you want to reload the file?') % (s, )
-                            dialog = MessageDialog(self.get_toplevel(), Gtk.MessageType.QUESTION, msg)
-                            ok = (dialog.run() == Gtk.ResponseType.OK)
-                            dialog.destroy()
-                            if ok:
-                                self.open_file(f, True)
-                except OSError:
-                    pass
-
```

Comment:
Why removing the function? Most of the time I tend to prefer to keep functions as it provides better code readability and maintainability, but **if** the reason is good enough, I'm ok to remove a function from time to time.

Reply:
To keep this a function, we could split it and make the first part of it a generator: we're collecting all changes before showing any confirmation dialog. Do you prefer that?

Reply:
This function was removed because it was empty after all of its code moved to another class.